silk/tests/openthread/ot_test_address_cache_table.py

```diff
@@ -0,0 +1,435 @@
+        self.wait_for_completion(self.device_list)
+
+        self.fed1.join(self.network_data, "end-node")
+        self.wait_for_completion(self.device_list)
+
+    @testcase.test_method_decorator
+    def test02_verify_node_type_and_parent(self):
+        self.assertTrue(self.r1.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_LEADER)
+        self.assertTrue(self.r2.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_ROUTER)
+        self.assertTrue(self.r3.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_ROUTER)
+        self.assertTrue(self.fed1.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_END_DEVICE)
+        self.assertTrue(self.sed2.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_SLEEPY_END_DEVICE)
+        self.assertTrue(self.fed3.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_END_DEVICE)
+
+        r1_ext_address = self.r1.getprop(wpan.WPAN_EXT_ADDRESS)[1:-1]
+        fed1_parent = self.fed1.getprop(wpan.WPAN_THREAD_PARENT)[1:17]
+        self.assertTrue(fed1_parent == r1_ext_address)
+
+        r2_ext_address = self.r2.getprop(wpan.WPAN_EXT_ADDRESS)[1:-1]
+        sed2_parent = self.sed2.getprop(wpan.WPAN_THREAD_PARENT)[1:17]
+        self.assertTrue(sed2_parent == r2_ext_address)
+
+        r3_ext_address = self.r3.getprop(wpan.WPAN_EXT_ADDRESS)[1:-1]
+        fed3_parent = self.fed3.getprop(wpan.WPAN_THREAD_PARENT)[1:17]
+        self.assertTrue(fed3_parent == r3_ext_address)
+
+    @testcase.test_method_decorator
+    def test03_verify_route_from_r1_to_r3(self):
+        # Wait till we have a valid "next hop" route on r1 towards r3
+        self.r1.add_prefix(PREFIX, stable=True, on_mesh=True, slaac=True, preferred=True)
+        self.wait_for_completion(self.device_list)
+
+        # Wait for prefix to get added to all the nodes
+        time.sleep(60)
+
+        self.get_rloc_and_ipv6_address_for_prefix()
+
+        def check_r1_router_table():
+            router_table = wpan_table_parser.parse_router_table_result(self.r1.get(wpan.WPAN_THREAD_ROUTER_TABLE))
+            self.assertEqual(len(router_table), 3)
+
+            for entry in router_table:
+                if entry.rloc16 == self.r3_rloc:
+                    verify(entry.next_hop != INVALID_ROUTER_ID)
+
+        verify_within(check_r1_router_table, ROUTER_TABLE_WAIT_TIME)
+
+    @testcase.test_method_decorator
+    def test04_verify_address_cache_table_on_r1(self):
+        # Send a single UDP message from r1 to sed2
+        # Send a single UDP message from r1 to fed3
+        self.transmit_receive_udp_message([(self.r1, self.sed2, self.r1_address, self.sed2_address),
+                                           (self.r1, self.fed3, self.r1_address, self.fed3_address)])
+        time.sleep(WAIT_TIME)
+        # The address cache table on r1 should contain two entries for sed2 and fed3 addresses.
+
+        addr_cache_table = self.r1.wpanctl("get", "get " + wpan.WPAN_THREAD_ADDRESS_CACHE_TABLE, 2)
+        addr_cache_table = wpan_table_parser.parse_address_cache_table_result(addr_cache_table)
```

Comment:
here parse_addess_cache_table is failing as out of 2 entries one entry is getting age:1 however the age:0 is required.

Reply:
Have you figured out what went wrong ?

Reply:
Not yet. The pairing and router table are proper and it shows all the links from r1 to r3. And it is not even related to delay testing by adding 180sec delay and that too did not affect the behavior. Digging more into it.

Reply:
Making changes for the following seems getting test_04 Pass.

class AddressCacheEntry(object):
     ....
      self._age = int(items_dict["Age"], 0)

def parse_address_cache_table_result(addr_cache_table_list):
    """Parses address cache table list string and returns an array of `AddressCacheEntry` objects.
    """
    return [AddressCacheEntry(item) for item in addr_cache_table_list.split("\n")[1:-2]]

Reply:
Thanks Qilin. Yes was indeed a parsing issue.